### PR TITLE
Update Glimmer VM

### PR DESCRIPTION
**Pull request by chadhietala**
This is the preemptive PR to integrate breaking changes from. https://github.com/glimmerjs/glimmer-vm/pull/784

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (krisselden) [inline .vscode/settings.json]: is this not the default?

**Comment by chadhietala** [comment]
Wanted to run a benchmark on this sense to get the tests passing I needed to bump the Testem timeout.

**TL;DR: Not stat-sig**

**Results**
screen shot 2018-03-05 at 3 58 02 pm

```

                          ██
                          ██
         ██               ██
       ████               ████
       ████               ████
       ████  ██           ██████
       ████  ██           ██████
       ████████           ██████
       ████████         ████████
     ██████████         ████████
     ██████████         ████████
     ██████████       ████████████
   ████████████       ████████████
     duration              js
pr-16268
40 samples
Chrome/64.0.3282.186
8 x Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz

          ms
type              P10         Q1     Median         Q3        P90        MAD
  duration 2675.21120 2689.28300 2703.20000 2730.94800 2744.06920   27.30801
  js       2125.62020 2142.67075 2160.26800 2177.92950 2189.78760   26.19977
          ms
type              IQR
  duration   41.66500
  js         35.25875

     ██
     ██
     ██                         ██
     ██                       ████
     ██                       ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ████
     ████                     ██████
   ████████          ██       ██████        ██
         duration                    js
master-8ef8dc63cb
40 samples
Chrome/64.0.3282.186
8 x Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz

          ms
type             P10        Q1    Median        Q3       P90       MAD
  duration 2668.7274 2687.0222 2697.3945 2719.6492 2729.1470   25.1827
  js       2119.5275 2136.4772 2152.7870 2169.6402 2184.6161   25.5667
          ms
type             IQR
  duration   32.6270
  js         33.1630

Test pr-16268 against master-8ef8dc63cb

	Wilcoxon rank sum test

data:  ms by set
W = 906, p-value = 0.3118
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 -5.734 18.373
sample estimates:
difference in location
                6.0235
```

**Review by rwjblue — APPROVED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (btecu) [reply .vscode/settings.json]: I believe that file is generated automatically by VS Code and should have not been committed & merged.

**Comment by chadhietala** [comment]
@krisselden I ran the bench again with the `Object.assign` fixes

Still no stat-sig.

screen shot 2018-03-06 at 9 16 04 am
screen shot 2018-03-06 at 9 16 14 am
screen shot 2018-03-06 at 9 16 21 am


```

           ██
           ██                         ██
           ████                     ████
           ████                     ████
           ████                     ████
           ████                     ████
           ██████                   ████
           ██████                 ██████
         ████████                 ██████  ██
         ████████                 ██████████
     ██  ████████████         ██████████████
   ██████████████████       ████████████████████
        duration                     js
pr-16268
40 samples
Chrome/64.0.3282.186
8 x Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz

          ms
type              P10         Q1     Median         Q3        P90        MAD
  duration 2650.03100 2662.00775 2669.85900 2680.02625 2689.68290   12.61174
  js       2096.60710 2109.86250 2120.01250 2129.50000 2148.98710   14.40717
          ms
type              IQR
  duration   18.01850
  js         19.63750

       ██
       ██
       ████
       ████
       ████
       ████
       ██████
       ██████                     ██████
       ████████                   ██████  ██
       ████████             ██    ██████  ██
       ████████             ██████████████████
     ██████████             ██████████████████
   ██████████████       ██  ██████████████████
      duration                    js
master-8ef8dc63cb
40 samples
Chrome/64.0.3282.186
8 x Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz

          ms
type              P10         Q1     Median         Q3        P90        MAD
  duration 2641.74000 2651.99700 2672.37300 2692.28075 2705.51570   30.40664
  js       2089.50960 2107.60825 2127.55800 2140.59375 2153.34530   25.79057
          ms
type              IQR
  duration   40.28375
  js         32.98550

Test pr-16268 against master-8ef8dc63cb

	Wilcoxon rank sum test

data:  ms by set
W = 697, p-value = 0.3258
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 -14.936   5.193
sample estimates:
difference in location
               -5.0895
```

**Comment by krisselden** [comment]
if you have it handy, I would run with the runtimeStats

**Pull request merged**